### PR TITLE
Parametrize test for __setitem__ (for dask array)

### PR DESCRIPTION
**Pull request by andersy005**
Follow up PR for #5174

- [x] Tests added
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Review by max-sixty — COMMENTED** [review]
We could avoid parametrizing the original array and the index value — these examples have them all the same, and we don't need to test over multiple values.

As a principle I'd vote to parametrize the minimum amount needed. At the extreme, we could put everything in some lisp-like data structure and the test just be `func(input) == expected`, but that actually makes it hard to understand.

**Comment by andersy005** [comment]
Thank you for the feedback, @max-sixty! I removed the redundant parameters... Let me know if there is anything else worth addressing...

File: xarray/tests/test_dask.py [hunk 1]
```diff
@@ -110,39 +110,40 @@ def test_indexing(self):
...
+        [
+            (
+                da.array([99, 2, 3, 4]),
+                0,
+            ),
+            (da.array([99, 99, 99, 4]), slice(2, None, -1)),
+            (da.array([99, 99, 3, 99]), [0, -1, 1]),
+            (da.array([99, 99, 99, 4]), np.arange(3)),
+            (da.array([1, 99, 99, 99]), [False, True, True, True]),
+            (da.array([1, 2, 3, 4]), da.array([1, 99, 99, 99]), np.arange(4) > 0),
```
Comment (max-sixty) [inline xarray/tests/test_dask.py:129]: Does this one still have three args in? Or am I misreading it?

**Review by max-sixty — APPROVED** [review]
Great! Thanks a lot @andersy005 !

File: xarray/tests/test_dask.py (hunk 1 above)
Comment (andersy005) [reply xarray/tests/test_dask.py]: Fixed...

**Comment by andersy005** [comment]
This should be mergeable once the CI is passing

**Comment by andersy005** [comment]
The flaky/failing test is due to an unstable OpenDAP server....

**Comment by mathause** [comment]
Thanks @andersy005!

**Pull request merged**